scripts/pin-github-actions.py

```
@@ -0,0 +1,167 @@
+
+
+def process_workflow_file(file_path: str, *, check_only: bool) -> None:
+    """Reads a file, replaces tags with hashes, and writes back the changes."""
+    with open(file_path, encoding="utf-8") as f:
+        old_content = content = f.read()
+
+    matches = ACTION_REGEX.findall(content)
+    if not matches:
+        return
+
+    for action_name, path, tag in matches:
```

Ruff (0.15.21)

[warning] Loop control variable `path` not used within loop body

Rename unused `path` to `_path`

(B007)